tests/functional/codegen/types/numbers/test_signed_ints.py

```diff
@@ -73,18 +74,35 @@ def foo(x: int256) -> int256:
 
 # TODO: make this test pass
 @pytest.mark.parametrize("base", (0, 1))
-def test_exponent_negative_power(get_contract, tx_failed, base):
+def test_exponent_negative_power_runtime_check(get_contract, tx_failed, base, experimental_codegen):
     # #2985
     code = f"""
 @external
-def bar() -> int16:
-    x: int16 = -2
+def bar(negative:int16) -> int16:
+    x: int16 = negative
     return {base} ** x
     """
     c = get_contract(code)
     # known bug: 2985
     with tx_failed():
-        c.bar()
+        c.bar(-2)
+
+
+@pytest.mark.parametrize("base", (0, 1))
+def test_exponent_negative_power_compile_time_check(
+    get_contract, tx_failed, base, experimental_codegen
+):
+    # #2985
+    code = f"""
+@external
+def bar() -> int16:
+    x: int16 = -2
+    return {base} ** x
+    """
+    if not experimental_codegen:
+        return
+    with pytest.raises(StaticAssertionException):
+        get_contract(code)
```

Comment:
i'm not sure we need the `compile_time_check` tests? or at least we should be testing the StaticAssertionException more specifically?